src/decode.rs

```diff
@@ -3495,13 +3492,19 @@ unsafe fn decode_sb(
     node: *const EdgeNode,
 ) -> Result<(), ()> {
     let ts = &mut *t.ts;
-    let hsz = 16 >> bl;
+    let hsz = 16 >> bl as u8;
     let have_h_split = f.bw > t.bx + hsz;
     let have_v_split = f.bh > t.by + hsz;
 
     if !have_h_split && !have_v_split {
-        assert!(bl < BL_8X8);
-        return decode_sb(c, t, f, bl + 1, (*(node as *const EdgeBranch)).split[0]);
+        assert!(bl != BlockLevel::BL_8X8);
```

Comment:
Since we are already doing things like `bl + 1`/`bl.decrease()`, I think deriving `Ord` is fine.  That said, if these are resolved by having `.decrease()` return an `Option` that we check (and thus do the `BL_8X8` check), that's fine, too.